core/primitives/src/bandwidth_scheduler.rs

```diff
@@ -301,6 +302,7 @@ impl BlockBandwidthRequests {
 /// The state should be the same on all shards. All shards start with the same state
```

Comment:
Random thought - can the state be kept off chain (e.g. by only storing the hash of it)? Would it make any difference?

Reply:
Chunk validators need the state to verify that the bandwidth grants are correct, so we'd have to provide that to them somehow :thinking:
In theory possible, but I don't think there are any advantages?